instance/models/server.py

```diff
@@ -282,7 +287,10 @@ def public_ip(self):
         if not self.openstack_id:
             return None
 
-        public_addr = openstack.get_server_public_address(self.os_server)
+        try:
+            public_addr = openstack.get_server_public_address(self.os_server)
+        except novaclient.exceptions.NotFound:
+            return None  # This server has been deleted.
```

Comment:
@bradenmacdonald Should this update the status of the server? For someone who doesn't know how the status of an app server is determined, it might be confusing to see app servers with no public IP and a "running" status listed in the API:

```
        ...
        "status": "ready",
        "public_ip": null,
        "status_name": "Ready",
        "status_description": "Booted and ready"
        ...
```

If you think we should address this, I'd be fine with doing it in a separate task.